SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h

```diff
@@ -176,27 +173,39 @@ class FReceptionistConfig : public FConnectionConfig
 
 	bool TryLoadCommandLineArgs()
 	{
-		bool bSuccess = true;
 		const TCHAR* CommandLine = FCommandLine::Get();
 
+		// Get command line options first since the URL handling will modify the CommandLine string
+		FParse::Value(CommandLine, TEXT("receptionistPort"), ReceptionistPort);
+
 		// Parse the command line for receptionistHost, if it exists then use this as the host IP.
-		if (!FParse::Value(CommandLine, TEXT("receptionistHost"), ReceptionistHost))
+		FString Host;
+		if (!FParse::Value(CommandLine, TEXT("receptionistHost"), Host))
 		{
 			// If a receptionistHost is not specified then parse for an IP address as the first argument and use this instead.
 			// This is how native Unreal handles connecting to other IPs, a map name can also be specified, in this case we use the default IP.
 			FString URLAddress;
-			FParse::Token(CommandLine, URLAddress, 0);
-			FRegexPattern Ipv4RegexPattern(TEXT("^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$"));
-			FRegexMatcher IpV4RegexMatcher(Ipv4RegexPattern, *URLAddress);
-			bSuccess = IpV4RegexMatcher.FindNext();
-			if (bSuccess)
+			FParse::Token(CommandLine, URLAddress, false);
+			const FURL URL(nullptr, *URLAddress, TRAVEL_Absolute);
+			if (URL.Valid)
 			{
-				SetReceptionistHost(URLAddress);
+				if (!URL.Host.IsEmpty())
+				{
+					SetReceptionistHost(URL.Host);
```

Comment:
Wonder if as part of SetReceptionistHost we can add the logic: 

if (URL.HasOption(*SpatialConstants::URL_USE_EXTERNAL_IP_FOR_BRIDGE_OPTION))
{
	UseExternalIp = true;
}

It seems everywhere we call SetReceptionistHost we all so do this logic after (ConnectionConfig.h - 196).

Reply:
True, unfortunately we need to keep another occurrence in `USpatialConnectionManager::SetupConnectionConfigFromURL` since we can have host-less URLs and there might be a use-case where we want to set this option but can only use the URL. Wondering if we need to support this use case, though?

Reply:
Added another comment on your new changes regarding this.